app/forms/openid_connect_token_form.rb

```diff
@@ -85,7 +85,7 @@ def validate_code_verifier
   def validate_client_assertion
     return unless identity.present?
 
-    service_provider = ServiceProvider.new(client_id)
+    service_provider = ServiceProvider.find_by_issuer!(client_id)
```

Comment:
Yeah I think so. New and unsaved AR models make me a little nervous somtimes (what if some code calls `.save!` on it?) so I would be in favor of an explicit `NullServiceProvider` class (similar to our `NullIdentity` class)